### DISCUSSION TITLE
In yolov5-5.0，Does correct[] count something less?

### DISCUSSION THREAD

**Discussion by snl-lzy** [ultralytics/yolov5#7079] [Q&A]
`                        # Append detections
                        detected_set = set()
                        for j in (ious > iouv[0]).nonzero(as_tuple=False):
                            d = ti[i[j]]  # detected target
                            if d.item() not in detected_set:
                                detected_set.add(d.item())
                                detected.append(d)
                                correct[pi[j]] = ious[j] > iouv  # iou_thres is 1xn
                                if len(detected) == nl:  # all targets already located in image
                                    break`
Consult the author, when a GT box corresponds to multiple prediction boxes, correct only counts the TP value of the first prediction box. Shouldn't we count the TP value of all prediction boxes?

**Comment by glenn-jocher** [comment]
@snl-lzy hi, thank you for your feature suggestion on how to improve YOLOv5 🚀!

The fastest and easiest way to incorporate your ideas into the official codebase is to submit a Pull Request (PR) implementing your idea, and if applicable providing before and after profiling/inference/training results to help us understand the improvement your feature provides. This allows us to directly see the changes in the code and to understand how they affect workflows and performance.

Please see our ✅ [Contributing Guide](https://github.com/ultralytics/yolov5/blob/master/CONTRIBUTING.md) to get started.

**Reply by snl-lzy** [reply]
Excuse me, I want to know if in yolov5-5.0, correct [] only counts the PT predicted for the first time. This problem has bothered me for a long time
